### ISSUE TITLE
input error for extractHairs

### ISSUE THREAD

**Issue by linzhangTuesday** [vibansal/HapCUT2#34]
have used HAPCUT before, used reference path directly following --ref, but this time with HAPCUT2/extractHAIRS, keep getting errors as below:
ERROR: Invalid input "--ref" for <0/1> option flag
also 
ERROR: Invalid number of arguments specified
trying to use HAPCUT for PacBio reads, the command line I am using are like
extractHAIRS --VCF sample.vcf --bam sample.sorted.bam --indels --ref path/to/ref.fasta --pacbio --out sample.fragment
have searched around but no answer found
Looking forward to your reply and many thanks!

**Comment by pjedge** [comment]
Change it to "--indels 1", it expects a 0/1 input after the flag. Also, use the --indels option with caution with PacBio reads. The phasing accuracy for indels is significantly worse for indels than for SNVs using error prone reads such as PacBio (with short indels such as 1 bp insertions being the worst).

**Comment by linzhangTuesday** [comment]
Hi Peter, Thank you very much for your quick responce!
Tried it with --indels 1 but got the below error
ERROR: Invalid number of arguments specified
thank you for reminding about the indel problems. the vcf I am using are derived from illumina reads while the bam files from Pacbio reads, hope that will make it better
Thanks again!

**Comment by pjedge** [comment]
Similarly, try changing "--pacbio" to "--pacbio 1"

**Comment by linzhangTuesday** [comment]
Thanks! it works!

**Comment by linzhangTuesday** [comment]
Thank you very much for all the suggestions, Peter.
this is the first time I am trying to construct haplotype from Pacbio reads. Could you please give me any other comments about the command line I am using? (maybe any other suggested options?)
Thanks!

**Comment by pjedge** [comment]
I think that it looks alright. Just note that the error rates of the shorter indels will be really unreasonably high, even if the variants are accurate (from illumina). The allelotyping of short indels with PacBio is very hard.

Be sure to filter your output based on the switch and mismatch quality scores in the last 2 columns of the haplotype output. This is very important for PacBio reads.

If you have Illumina reads anyway, you might want to extract those as well using extracthairs, and then concatenate the Illumina fragment file to the PacBio fragment file. It probably won't do much but it might help to make the local phase of dense variant clusters (that are close enough to phase with Illumina) more accurate.

**Comment by pjedge** [comment]
Note: if the pacbio read coverage is lower than ~40x you might be better off just leaving --indels off of the PacBio extractHairs command (you can leave it on for the illumina extractHairs command, and you'll still phase a few indels that can be linked to a nearby SNV by an illumina read).

**Comment by linzhangTuesday** [comment]
Hi Peter, thank you very much! Sorry I was not feeling well for the past few days so didn't check the updates.
To make use of both illumina and Pacbio reads, may I ask do you mean use cat to concatenate the illumina and Pacbio fragment files? 
For filtering switch and mismatch quality scores, I found the --error_analysis_mode option in hapcut which says "compute switch confidence scores and print to haplotype file but don't split blocks or prune” but no option for mismatch scores. Were you talking about filter them in downstream analysis? 

Besides, I also have Hi-C reads for some samples and I noticed there were Hi-C options for HAPCUT2, do you suggest using Hi-C reads for extractHAIRS then concatenate them too with illumina and PacBio reads? But I also noticed there is a pipeline designed especially for combine Hi-C and PacBio https://github.com/vibansal/HapCUT2/tree/master/recipes/HiC_Longread

**Comment by pjedge** [comment]
Yes, concatenating the files with "cat" should do the trick.
Yes, --error_analysis_mode 1 will produce the switch error quality scores. The mismatch scores are already produced by default. If you use --error_analysis_mode 1, then you can filter the resulting file with:
```
python3 utilities/prune_haplotypes.py -i input.hap -o output.hap -mq 90 -sq 30
```
This says "prune out SNVs with mismatch quality less than 90 and split haplotype blocks if the switch quality is below 30". The numbers I provided are just examples, but I would consider them good cutoffs. If you have low read coverage, then it might prune a bit too much.

**Comment by linzhangTuesday** [comment]
Hi Peter, thank you very much for the detailed instructions! Have tried prune_haplotypes.py on the illumina output (the Pacbio one is still running). Will try to cat them once done.
Thank you again! Your suggestions have made my life much easier!

**Comment by pjedge** [comment]
That doesn't sound right. Just so we're clear, you should run extracthairs twice, once on Illumina and once on PacBio. Then you should concatenate the two resulting fragment files into one fragment file. Then use the concatenated fragment file as input to HapCUT2. Then you should run prune_haplotypes.py on the output of HapCUT2 (only once).

The prune haplotypes script needs to be run on the output haplotypes from HapCUT2, not on the fragment files.

I apologise that the process is unnecessarily confusing, especially for multiple combined input read types. We're planning on wrapping all these steps into a one-step python executable, but we haven't had time to do it yet.

**Comment by linzhangTuesday** [comment]
Thanks a lot for the nice clarification, Peter. : )
I accidentally launched both extractHAIRS and HAPCUT2 for both illumina and Pacbio using slurm for multiple samples of the same organism (probably that just meant I was not quite clear about what to do). But did use prune script on haplotye file ;P ...
My plan is to finish these following your instructions and later I will try to learn to implement Hi-C reads for some of these samples.

**Comment by linzhangTuesday** [comment]
but I did use different slurm file so --indel 1 was not used for PacBio reads : )
